### PR TITLE
Fix extensions error

### PR DESCRIPTION
**Pull request by ClementPasteau**
Introduced by #6200

### CODE REVIEW

**Review by AlexandreSi — APPROVED** [review]

**Comment by arthuro555** [comment]
Did you manually change GDevelop.js/types.d.ts? That file is auto generated so the changes will not remain, rather the generation script at `GDevelop.js/scripts/generate-dts.mjs` should be altered.

**Comment by 4ian** [comment]
Should we:
- fix all the errors (https://gdevelop.semaphoreci.com/jobs/fe013e6f-0e2a-48d0-92c6-7f063370778e)
- or, for the sake of unlocking the release, going back at ignoring them (while we wait for a PR to fix them while NOT breaking the extensions :))?

A lot of them can be ts-ignored, but some are interesting and might be an error either in typing or in the actual code using them:

image

In any case we want *no regression* in the extensions because the new release will start soon.

**Comment by ClementPasteau** [comment]
> Did you manually change GDevelop.js/types.d.ts? That file is auto generated so the changes will not remain, rather the generation script at `GDevelop.js/scripts/generate-dts.mjs` should be altered.

Yep I did as a test and realised it was auto-generated afterwards, this has to be fixed in the script indeed

**Comment by ClementPasteau** [comment]
A lot of errors are coming from optional parameters of functions.
image

Unfortunately, the `Bindings.idl` does not contain the information about which parameters are optional, so unsure how we can safely generate ts types from this.

I'm suggesting to revert for the moment, so we can release, and work on a fix for all files for next one.

**Comment by arthuro555** [comment]
I'm certain I implemented optional parameters and default values in the parser, and that there was a way to declare parameters as optional in WebIDL. Simply add the word optional as you would for example add unsigned before a type. See here: https://github.com/4ian/GDevelop/blob/dd090fd1d7df5fd7e3f48f31df95e8d6e1f2afa9/GDevelop.js/Bindings/Bindings.idl#L2302

**Comment by 4ian** [comment]
Emscripten does not support optional parameters if I remember correctly, but it "worked" when `undefined` was passed so we indeed hardcoded some tolerance in the Flow types of some functions.

File: GDevelop.js/scripts/generate-dts.mjs [hunk 1]
```diff
@@ -175,7 +175,9 @@ for (const [_, interfaceName, interfaceCode] of bindingsFile.matchAll(
 
     if (isAttribute) {
       const attributeName = Parser.readUntil(';');
-      attributes.push(`${attributeName}: ${returnType};`);
+      attributes.push(
+        `${attributeName}${optionalReturn ? '?' : ''}: ${returnType};`
```
Comment (arthuro555) [inline GDevelop.js/scripts/generate-dts.mjs:179]: I don't think this makes sense, class properties cannot be optional in c++ after all

**Comment by ClementPasteau** [comment]
I'm reverting #6200 because there seems to be some leftovers from it that cannot be fixed quickly.
Please re-submit a PR with the types errors fixed @arthuro555 we'd love that :) And feel free to look at what I started here
If you don't have time we may take a look at it later!

https://github.com/4ian/GDevelop/pull/6220

**Pull request closed without merging**